File: rules-interpreter/src/main/java/com/github/iamrenny/ruleflow/evaluators/RegexContextEvaluator.java
package com.github.iamrenny.ruleflow.evaluators;

import com.github.iamrenny.ruleflow.RuleFlowLanguageParser.RegexlikeContext;
import com.github.iamrenny.ruleflow.visitors.Visitor;

import java.util.regex.Pattern;

public class RegexContextEvaluator implements ContextEvaluator<RegexlikeContext> {

    @Override
    public Object evaluate(RegexlikeContext ctx, Visitor visitor) {
        Object value = visitor.visit(ctx.value);

        String regexPattern = ctx.regex.getText().replace("'", "");
        String input = value.toString();

        Pattern pattern = Pattern.compile(regexPattern);
        return pattern.matcher(input).replaceAll("");
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/analang/evaluators/RegexContextEvaluator.kt
package com.github.iamrenny.rulesflow.evaluators

import com.rappi.fraud.analang.ANAParser.RegexlikeContext
import com.github.iamrenny.rulesflow.visitors.Visitor

class RegexContextEvaluator: ContextEvaluator<RegexlikeContext> {
    override fun evaluate(ctx: RegexlikeContext, visitor: Visitor): Any {
        val value = visitor.visit(ctx.value)


        return Regex(ctx.regex.text.replace("'", "")).replace(value.toString(), "")
    }
}